packages/react-widgets/src/widgets/TimeSeriesWidget.js

```diff
@@ -0,0 +1,303 @@
+import React, { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { getTimeSeries } from '../models';
+import { addFilter, removeFilter, selectSourceById } from '@carto/react-redux';
+import { NoDataAlert, TimeSeriesWidgetUI, WrapperWidgetUI } from '@carto/react-ui';
+import {
+  GroupDateTypes,
+  AggregationTypes,
+  _FilterTypes as FilterTypes,
+  _getApplicableFilters as getApplicableFilters
+} from '@carto/react-core';
+import { capitalize, Menu, MenuItem, SvgIcon, Typography } from '@material-ui/core';
+import { TIME_SERIES_CHART_TYPES } from '@carto/react-ui';
+import { PropTypes } from 'prop-types';
+
+const BUCKET_SIZE_RANGE_MAPPING = {
+  [GroupDateTypes.WEEKS]: 60 * 60 * 24 * 7,
+  [GroupDateTypes.DAYS]: 60 * 60 * 24
+};
+
+function TimeSeriesWidget({
+  // Widget
+  id,
+  title,
+  dataSource,
+  column,
+  operationColumn,
+  operation,
+  stepSizeOptions,
+  onError,
+  wrapperProps,
+  noDataAlertProps,
+  // UI
+  chartType,
+  duration,
+  tooltip,
+  tooltipFormatter,
+  formatter,
+  height,
+  showControls,
+  isPlaying,
+  onPlay,
+  isPaused,
+  onPause,
+  onStop,
+  timelinePosition,
+  onTimelineUpdate,
+  timeframe,
+  onTimeframeUpdate,
+  // Both
+  stepSize
+}) {
+  const dispatch = useDispatch();
+  const [timeSeriesData, setTimeSeriesData] = useState([]);
+  const viewportFeaturesReady = useSelector((state) => state.carto.viewportFeaturesReady);
+  const [selectedStepSize, setSelectedStepSize] = useState(stepSize);
+
+  useEffect(() => {
+    if (stepSize !== selectedStepSize) {
+      setSelectedStepSize(stepSize);
+    }
+    // Only on bucketSize update
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [stepSize]);
+
+  const [isLoading, setIsLoading] = useState(false);
+
+  const filters = useSelector((state) => {
+    const source = selectSourceById(state, dataSource) || {};
+    const filters = { ...source.filters };
+    delete filters[column];
+    return filters;
+  });
+
+  const isSourceReady = !!viewportFeaturesReady[dataSource];
+
+  useEffect(() => {
+    if (isSourceReady) {
+      setIsLoading(true);
```

Comment:
The same reason as in my first comment. If I don't control the loading by myself, the component will be always loading because each time a filter is applied all the widgets loading state are set to true because the loading state is centralised, and each time the source changes, every widget fed by that source is called to load (changing isLoading from false to true). I don't think it's good.

Reply:
not applicable now